### PR TITLE
[FAB-17238] Refactor couchdb test in ./core/ledger/

### PR DESCRIPTION
**Pull request by DereckLuo**
Signed-off-by: Chongxin Luo <Chongxin.Luo@ibm.com>

#### Type of change

- Improvement (improvement to code, performance, etc)

#### Description

- Refactor couchdb tests in ./core/ledger/... Only starting couchDB container
ones for all couchdb unit tests.
- Fix couchdb containers not being cleaned up after tests complete.

#### Related issues

https://jira.hyperledger.org/browse/FAB-17238

### CODE REVIEW

File: core/ledger/util/couchdbtest/couchdbTest.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/*
```
Comment (mastersingh24) [inline core/ledger/util/couchdbtest/couchdbTest.go]: we don't really use camelCase for filenames.  I'd justname the file `couchdb.go`

File: core/ledger/util/couchdbtest/couchdbTest.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		}
+	}
+	err := couchDB.Start()
+	if err != nil {
+		err = fmt.Errorf("failed to start couchDB : %s", err)
+		panic(err)
+	}
+
+	os.Setenv("COUCHDB_ADDR", couchDB.Address())
+	return couchDB.Address(), func() { couchDB.Stop() }
```
Comment (mastersingh24) [inline core/ledger/util/couchdbtest/couchdbTest.go]:
```suggestion
return couchDB.Address(), func() {
     couchDB.Stop()
     os.Unsetenv("COUCHDB_ADDR")
}

File: core/ledger/util/couchdbtest/couchdbTest.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+// CouchDBCleanup cleans up external couchDB resource
```
Comment (mastersingh24) [inline core/ledger/util/couchdbtest/couchdbTest.go]: I'd remove ... see suggestion above

**Review by mastersingh24 — CHANGES_REQUESTED** [review]

**Review by mastersingh24 — COMMENTED** [review]
Looks good overall ... just a couple of suggestions

**Comment by DereckLuo** [comment]
thanks for the review, changes has been applied. it is much cleaner than original solution.

**Review by mastersingh24 — APPROVED** [review]

**Comment by mastersingh24** [comment]
I'm overriding this ... another gossip flake ... not going to waste another CI cycle

**Pull request merged**